torchgeo/datasets/eurocrops.py

```diff
@@ -0,0 +1,270 @@
+        ("HR_2020.zip", "8bfe2b0cbd580737adcf7335682a1ea5"),
+        ("LT_2021.zip", "c7597214b90505877ee0cfa1232ac45f"),
+        ("LV_2021.zip", "b7253f96c8699d98ca503787f577ce26"),
+        ("NL_2020.zip", "823da32d28695b8b016740449391c0db"),
+        ("PT.zip", "3dba9c89c559b34d57acd286505bcb66"),
+        ("SE_2021.zip", "cab164c1c400fce56f7f1873bc966858"),
+        ("SI_2021.zip", "6b2dde6ba9d09c3ef8145ea520576228"),
+        ("SK_2021.zip", "c7762b4073869673edc08502e7b22f01"),
+        # Year is unknown for Romania portion (ny = no year).
+        # We skip since it is inconsistent with the rest of the data.
+        # ("RO_ny.zip", "648e1504097765b4b7f825decc838882"),
+    ]
+
+    def __init__(
+        self,
+        paths: Union[str, Iterable[str]] = "data",
+        crs: CRS = CRS.from_epsg(4326),
+        res: float = 0.00001,
+        classes: Optional[list[str]] = None,
+        transforms: Optional[Callable[[dict[str, Any]], dict[str, Any]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new EuroCrops instance.
+
+        Args:
+            paths: one or more root directories to search for files to load
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to WGS-84)
+            res: resolution of the dataset in units of CRS
+            classes: list of classes to include (specified by their HCAT code),
+                the rest will be mapped to 0 (defaults to all classes)
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        self.paths = paths
+        self.checksum = checksum
+
+        if download:
+            self._download()
+
+        if not self._check_integrity():
+            raise DatasetNotFoundError(self)
+
+        self._load_class_map(classes)
+
+        super().__init__(
+            paths=paths,
+            crs=crs,
+            res=res,
+            transforms=transforms,
+            label_name=self.label_name,
+        )
+
+    def _check_integrity(self) -> bool:
+        """Check integrity of dataset.
+
+        Returns:
+            True if dataset files are found and/or MD5s match, else False
+        """
+        assert isinstance(self.paths, str)
+
+        filepath = os.path.join(self.paths, self.hcat_fname)
+        if not check_integrity(filepath, self.hcat_md5 if self.checksum else None):
+            return False
+
+        for fname, md5 in self.zenodo_files:
+            filepath = os.path.join(self.paths, fname)
+            if not check_integrity(filepath, md5 if self.checksum else None):
+                return False
```

Comment:
This is the only remaining line not covered by unit tests

Reply:
Fixed!